tests/test_virtualenv.py

```diff
@@ -538,6 +539,28 @@ def test_create_reuse_uv_environment(make_one):
     assert reused
 
 
+UV_IN_PIPX_VENV = "/home/user/.local/pipx/venvs/nox/bin/uv"
+
+
+@pytest.mark.parametrize(
+    ["which_result", "find_uv_bin_result", "expected"],
+    [
+        ("/usr/bin/uv", UV_IN_PIPX_VENV, "/usr/bin/uv"),
```

Comment:
Is this the desired behavior?

You might argue that Nox should prefer the `uv` in its own environment.

Or you might argue that we should prefer `PATH` lookup because it's the standard mechanism, and looking in the environment is just a fallback for an edge case.

We're taking the second stance here, but I don't have a strong opinion fwiw.

Reply:
I'd say principal of least surprise suggests the second is best IMO

Reply:
In scikit-build-core, I try to import CMake/ninja first before falling back on path search. For a Python library, using something coinstalled I think makes more sense. Otherwise you can’t pip install uv to get the latest uv if you have a system uv.

Reply:
Provided that the venv is activated, uv from the venv's bin would take precedence, so `pip install uv` would work. pipx is a bit of an edge case, and an argument could probably be made that it should expose its venv/bin folder to its exported scripts.

Reply:
Not sure why it should be assumed that it's active? Usually environments are designed to be used without being active (though there are some exceptions).

Right now, it might not seem to make much difference, but as uv gets older and pre-installed in more places, it's likely to be much more important to be able to pip install a modern uv. That's really important for cmake/ninja (more for cmake since it changes more).

Reply:
I'm fine to reverse the order.

How do you actually upgrade uv in Nox's pipx environment? AFAIU `pipx upgrade nox` wouldn't touch dependencies, and `--include-injected` doesn't apply. So would you need to `pipx runpip nox install --upgrade uv`? This seems sufficiently obscure that most users would never upgrade uv. We may want to document this, if there's no better way.

Reply:
I just checked, and the only way I found to upgrade uv was the `runpip` method.

To be honest, I think we should drop the extra. I fear that most users would end up with an outdated uv. They're better off installing uv explicitly, like with `pipx install nox uv`. Then `pipx upgrade-all` will just work.

Reply:
If uv isn't declared as a dependency of nox, there shouldn't be an expectation that nox would find ux from anywhere that's not on `$PATH` - could simply drop the import?

Reply:
To be clear, here's what I'm proposing:

- close this PR (which means that nox will only look for uv on PATH)
- open a new PR that removes the `nox[uv]` extra

Reply:
We can place the minimum version we require in the extra. Then it will get upgraded if it’s too old for us. You can also use pipx’s reinstall all.

The extra is useful to make this work in the action.

Reply:
> We can place the minimum version we require in the extra. Then it will get upgraded if it’s too old for us.

I'm not worried about our requirements, but about those of our users when calling `session.install`. Also, Nox releases less frequently than uv, by far.

> You can also use pipx reinstall nox[uv].

That's not a good experience for users. People will use `pipx install nox[uv]` and `pipx upgrade-all`, and it shouldn't leave them with an outdated uv.

> Pipx run (which is almost exclusively what I use) reinstalls after a week.

I don't think the majority of users uses `pipx run nox`. For those injecting packages like `nox-poetry` that's not an option.

> The extra is useful to make this work in the action.

If there's another way to make this work in the action that doesn't cause problems for our users and support overhead for us, I'd say that's worth investigating.

Reply:
Related: https://github.com/pypa/pipx/issues/925

Reply:
It looks like this might be very easily fixable from pipx's side (see linked issue).

Reply:
Ok I'm fine to go ahead with this here.